bcda/cclf/local_fileprocessor.go

```diff
@@ -280,3 +281,61 @@ func (processor *LocalFileProcessor) OpenZipArchive(filePath string) (*zip.Reade
 		}
 	}, err
 }
+
+func (processor *LocalFileProcessor) CleanUpCSV(file csvFile) error {
+	var err error
+
+	func() {
+		close := metrics.NewChild(context.Background(), "cleanUpCSV")
+		defer close()
+
+		processor.Handler.Logger.Infof("Cleaning up file %s.\n", file.metadata.name)
+		folderName := filepath.Base(file.filepath)
+		newpath := fmt.Sprintf("%s/%s", conf.GetEnv("PENDING_DELETION_DIR"), folderName)
+		if !file.imported {
+			// check the timestamp on the failed files
+			elapsed := time.Since(file.metadata.deliveryDate).Hours()
+			deleteThreshold := utils.GetEnvInt("BCDA_ETL_FILE_ARCHIVE_THRESHOLD_HR", 72)
+			if int(elapsed) > deleteThreshold {
+				if _, err = os.Stat(newpath); err == nil {
+					return
+				}
+				// move the (un)successful files to the deletion dir
+				err = os.Rename(file.filepath, newpath)
+				if err != nil {
+					processor.Handler.Logger.Errorf("File %s failed to clean up properly: %v", file.filepath, err)
+				} else {
+					processor.Handler.Logger.Infof("File %s never ingested, moved to the pending deletion dir", file.filepath)
+				}
+			}
+		} else {
+			if _, err = os.Stat(newpath); err == nil {
+				return
+			}
+			// move the successful file to the deletion dir
+			err := os.Rename(file.filepath, newpath)
+			if err != nil {
+				processor.Handler.Logger.Errorf("File %s failed to clean up properly: %v", file.filepath, err)
+
+			} else {
+				processor.Handler.Logger.Infof("File %s successfully ingested, moved to the pending deletion dir", file.filepath)
+			}
+		}
+	}()
+	return err
+}
+
+func (processor *LocalFileProcessor) LoadCSV(filepath string) (*bytes.Reader, func(), error) {
```

Comment:
we had old infrastructure -- using this allowed us to keep the old while we implemented thew new. This will also allow us to implement a 'processor' for local testing.

Reply:
I think it would be best to try to mimic prod processing but understandable if its convenient in other ways.

Reply:
yeah, I think we should look to see if this is still necessary. I can do some investigation -- I thought this was perhaps used during our CI/CD, but I could be wrong.